// @pulumi/azure-native/types/enums/servicelinker/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VNetSolutionType = exports.TargetServiceType = exports.SecretType = exports.DryrunActionName = exports.DeleteOrUpdateBehavior = exports.ClientType = exports.AzureResourceType = exports.AuthType = exports.AllowType = exports.ActionType = exports.AccessKeyPermissions = exports.v20230401preview = exports.v20221101preview = exports.v20211101preview = void 0;
// Export sub-modules:
const v20211101preview = require("./v20211101preview");
exports.v20211101preview = v20211101preview;
const v20221101preview = require("./v20221101preview");
exports.v20221101preview = v20221101preview;
const v20230401preview = require("./v20230401preview");
exports.v20230401preview = v20230401preview;
exports.AccessKeyPermissions = {
    Read: "Read",
    Write: "Write",
    Listen: "Listen",
    Send: "Send",
    Manage: "Manage",
};
exports.ActionType = {
    Enable: "enable",
    OptOut: "optOut",
};
exports.AllowType = {
    True: "true",
    False: "false",
};
exports.AuthType = {
    SystemAssignedIdentity: "systemAssignedIdentity",
    UserAssignedIdentity: "userAssignedIdentity",
    ServicePrincipalSecret: "servicePrincipalSecret",
    ServicePrincipalCertificate: "servicePrincipalCertificate",
    Secret: "secret",
    AccessKey: "accessKey",
    UserAccount: "userAccount",
};
exports.AzureResourceType = {
    KeyVault: "KeyVault",
};
exports.ClientType = {
    None: "none",
    Dotnet: "dotnet",
    Java: "java",
    Python: "python",
    Go: "go",
    Php: "php",
    Ruby: "ruby",
    Django: "django",
    Nodejs: "nodejs",
    SpringBoot: "springBoot",
    Kafka_springBoot: "kafka-springBoot",
    Dapr: "dapr",
};
exports.DeleteOrUpdateBehavior = {
    Default: "Default",
    ForcedCleanup: "ForcedCleanup",
};
exports.DryrunActionName = {
    CreateOrUpdate: "createOrUpdate",
};
exports.SecretType = {
    RawValue: "rawValue",
    KeyVaultSecretUri: "keyVaultSecretUri",
    KeyVaultSecretReference: "keyVaultSecretReference",
};
exports.TargetServiceType = {
    AzureResource: "AzureResource",
    ConfluentBootstrapServer: "ConfluentBootstrapServer",
    ConfluentSchemaRegistry: "ConfluentSchemaRegistry",
    SelfHostedServer: "SelfHostedServer",
};
exports.VNetSolutionType = {
    ServiceEndpoint: "serviceEndpoint",
    PrivateLink: "privateLink",
};
