// @pulumi/azure-native/appcomplianceautomation/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20221116preview = exports.Report = exports.getReportOutput = exports.getReport = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getReport = null;
exports.getReportOutput = null;
utilities.lazyLoad(exports, ["getReport", "getReportOutput"], () => require("./getReport"));
exports.Report = null;
utilities.lazyLoad(exports, ["Report"], () => require("./report"));
// Export sub-modules:
const v20221116preview = require("./v20221116preview");
exports.v20221116preview = v20221116preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:appcomplianceautomation:Report":
                return new exports.Report(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "appcomplianceautomation", _module);
