// @pulumi/azure-native/storagecache/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20231101preview = exports.v20230501 = exports.v20230301preview = exports.v20210301 = exports.StorageTarget = exports.getStorageTargetOutput = exports.getStorageTarget = exports.getRequiredAmlFSSubnetsSizeOutput = exports.getRequiredAmlFSSubnetsSize = exports.getCacheOutput = exports.getCache = exports.getAmlFilesystemOutput = exports.getAmlFilesystem = exports.Cache = exports.AmlFilesystem = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AmlFilesystem = null;
utilities.lazyLoad(exports, ["AmlFilesystem"], () => require("./amlFilesystem"));
exports.Cache = null;
utilities.lazyLoad(exports, ["Cache"], () => require("./cache"));
exports.getAmlFilesystem = null;
exports.getAmlFilesystemOutput = null;
utilities.lazyLoad(exports, ["getAmlFilesystem", "getAmlFilesystemOutput"], () => require("./getAmlFilesystem"));
exports.getCache = null;
exports.getCacheOutput = null;
utilities.lazyLoad(exports, ["getCache", "getCacheOutput"], () => require("./getCache"));
exports.getRequiredAmlFSSubnetsSize = null;
exports.getRequiredAmlFSSubnetsSizeOutput = null;
utilities.lazyLoad(exports, ["getRequiredAmlFSSubnetsSize", "getRequiredAmlFSSubnetsSizeOutput"], () => require("./getRequiredAmlFSSubnetsSize"));
exports.getStorageTarget = null;
exports.getStorageTargetOutput = null;
utilities.lazyLoad(exports, ["getStorageTarget", "getStorageTargetOutput"], () => require("./getStorageTarget"));
exports.StorageTarget = null;
utilities.lazyLoad(exports, ["StorageTarget"], () => require("./storageTarget"));
// Export enums:
__exportStar(require("../types/enums/storagecache"), exports);
// Export sub-modules:
const v20210301 = require("./v20210301");
exports.v20210301 = v20210301;
const v20230301preview = require("./v20230301preview");
exports.v20230301preview = v20230301preview;
const v20230501 = require("./v20230501");
exports.v20230501 = v20230501;
const v20231101preview = require("./v20231101preview");
exports.v20231101preview = v20231101preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:storagecache:AmlFilesystem":
                return new exports.AmlFilesystem(name, undefined, { urn });
            case "azure-native:storagecache:Cache":
                return new exports.Cache(name, undefined, { urn });
            case "azure-native:storagecache:StorageTarget":
                return new exports.StorageTarget(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "storagecache", _module);
