// @pulumi/azure-native/machinelearningcompute/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20170801preview = exports.OperationalizationCluster = exports.listOperationalizationClusterKeysOutput = exports.listOperationalizationClusterKeys = exports.getOperationalizationClusterOutput = exports.getOperationalizationCluster = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getOperationalizationCluster = null;
exports.getOperationalizationClusterOutput = null;
utilities.lazyLoad(exports, ["getOperationalizationCluster", "getOperationalizationClusterOutput"], () => require("./getOperationalizationCluster"));
exports.listOperationalizationClusterKeys = null;
exports.listOperationalizationClusterKeysOutput = null;
utilities.lazyLoad(exports, ["listOperationalizationClusterKeys", "listOperationalizationClusterKeysOutput"], () => require("./listOperationalizationClusterKeys"));
exports.OperationalizationCluster = null;
utilities.lazyLoad(exports, ["OperationalizationCluster"], () => require("./operationalizationCluster"));
// Export enums:
__exportStar(require("../types/enums/machinelearningcompute"), exports);
// Export sub-modules:
const v20170801preview = require("./v20170801preview");
exports.v20170801preview = v20170801preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:machinelearningcompute:OperationalizationCluster":
                return new exports.OperationalizationCluster(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "machinelearningcompute", _module);
