// @pulumi/azure-native/machinelearningexperimentation/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20170501preview = exports.Workspace = exports.Project = exports.getWorkspaceOutput = exports.getWorkspace = exports.getProjectOutput = exports.getProject = exports.getAccountOutput = exports.getAccount = exports.Account = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Account = null;
utilities.lazyLoad(exports, ["Account"], () => require("./account"));
exports.getAccount = null;
exports.getAccountOutput = null;
utilities.lazyLoad(exports, ["getAccount", "getAccountOutput"], () => require("./getAccount"));
exports.getProject = null;
exports.getProjectOutput = null;
utilities.lazyLoad(exports, ["getProject", "getProjectOutput"], () => require("./getProject"));
exports.getWorkspace = null;
exports.getWorkspaceOutput = null;
utilities.lazyLoad(exports, ["getWorkspace", "getWorkspaceOutput"], () => require("./getWorkspace"));
exports.Project = null;
utilities.lazyLoad(exports, ["Project"], () => require("./project"));
exports.Workspace = null;
utilities.lazyLoad(exports, ["Workspace"], () => require("./workspace"));
// Export sub-modules:
const v20170501preview = require("./v20170501preview");
exports.v20170501preview = v20170501preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:machinelearningexperimentation:Account":
                return new exports.Account(name, undefined, { urn });
            case "azure-native:machinelearningexperimentation:Project":
                return new exports.Project(name, undefined, { urn });
            case "azure-native:machinelearningexperimentation:Workspace":
                return new exports.Workspace(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "machinelearningexperimentation", _module);
