// @pulumi/azure-native/autonomousdevelopmentplatform/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20211101preview = exports.getDataPoolOutput = exports.getDataPool = exports.getAccountOutput = exports.getAccount = exports.DataPool = exports.Account = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Account = null;
utilities.lazyLoad(exports, ["Account"], () => require("./account"));
exports.DataPool = null;
utilities.lazyLoad(exports, ["DataPool"], () => require("./dataPool"));
exports.getAccount = null;
exports.getAccountOutput = null;
utilities.lazyLoad(exports, ["getAccount", "getAccountOutput"], () => require("./getAccount"));
exports.getDataPool = null;
exports.getDataPoolOutput = null;
utilities.lazyLoad(exports, ["getDataPool", "getDataPoolOutput"], () => require("./getDataPool"));
// Export enums:
__exportStar(require("../types/enums/autonomousdevelopmentplatform"), exports);
// Export sub-modules:
const v20211101preview = require("./v20211101preview");
exports.v20211101preview = v20211101preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:autonomousdevelopmentplatform:Account":
                return new exports.Account(name, undefined, { urn });
            case "azure-native:autonomousdevelopmentplatform:DataPool":
                return new exports.DataPool(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "autonomousdevelopmentplatform", _module);
