// @pulumi/azure-native/mobilepacketcore/v20231015preview/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpfDeployment = exports.SmfDeployment = exports.ObservabilityService = exports.NssfDeployment = exports.NrfDeployment = exports.getUpfDeploymentOutput = exports.getUpfDeployment = exports.getSmfDeploymentOutput = exports.getSmfDeployment = exports.getObservabilityServiceOutput = exports.getObservabilityService = exports.getNssfDeploymentOutput = exports.getNssfDeployment = exports.getNrfDeploymentOutput = exports.getNrfDeployment = exports.getClusterServiceOutput = exports.getClusterService = exports.getAmfDeploymentOutput = exports.getAmfDeployment = exports.ClusterService = exports.AmfDeployment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.AmfDeployment = null;
utilities.lazyLoad(exports, ["AmfDeployment"], () => require("./amfDeployment"));
exports.ClusterService = null;
utilities.lazyLoad(exports, ["ClusterService"], () => require("./clusterService"));
exports.getAmfDeployment = null;
exports.getAmfDeploymentOutput = null;
utilities.lazyLoad(exports, ["getAmfDeployment", "getAmfDeploymentOutput"], () => require("./getAmfDeployment"));
exports.getClusterService = null;
exports.getClusterServiceOutput = null;
utilities.lazyLoad(exports, ["getClusterService", "getClusterServiceOutput"], () => require("./getClusterService"));
exports.getNrfDeployment = null;
exports.getNrfDeploymentOutput = null;
utilities.lazyLoad(exports, ["getNrfDeployment", "getNrfDeploymentOutput"], () => require("./getNrfDeployment"));
exports.getNssfDeployment = null;
exports.getNssfDeploymentOutput = null;
utilities.lazyLoad(exports, ["getNssfDeployment", "getNssfDeploymentOutput"], () => require("./getNssfDeployment"));
exports.getObservabilityService = null;
exports.getObservabilityServiceOutput = null;
utilities.lazyLoad(exports, ["getObservabilityService", "getObservabilityServiceOutput"], () => require("./getObservabilityService"));
exports.getSmfDeployment = null;
exports.getSmfDeploymentOutput = null;
utilities.lazyLoad(exports, ["getSmfDeployment", "getSmfDeploymentOutput"], () => require("./getSmfDeployment"));
exports.getUpfDeployment = null;
exports.getUpfDeploymentOutput = null;
utilities.lazyLoad(exports, ["getUpfDeployment", "getUpfDeploymentOutput"], () => require("./getUpfDeployment"));
exports.NrfDeployment = null;
utilities.lazyLoad(exports, ["NrfDeployment"], () => require("./nrfDeployment"));
exports.NssfDeployment = null;
utilities.lazyLoad(exports, ["NssfDeployment"], () => require("./nssfDeployment"));
exports.ObservabilityService = null;
utilities.lazyLoad(exports, ["ObservabilityService"], () => require("./observabilityService"));
exports.SmfDeployment = null;
utilities.lazyLoad(exports, ["SmfDeployment"], () => require("./smfDeployment"));
exports.UpfDeployment = null;
utilities.lazyLoad(exports, ["UpfDeployment"], () => require("./upfDeployment"));
// Export enums:
__exportStar(require("../../types/enums/mobilepacketcore/v20231015preview"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:mobilepacketcore/v20231015preview:AmfDeployment":
                return new exports.AmfDeployment(name, undefined, { urn });
            case "azure-native:mobilepacketcore/v20231015preview:ClusterService":
                return new exports.ClusterService(name, undefined, { urn });
            case "azure-native:mobilepacketcore/v20231015preview:NrfDeployment":
                return new exports.NrfDeployment(name, undefined, { urn });
            case "azure-native:mobilepacketcore/v20231015preview:NssfDeployment":
                return new exports.NssfDeployment(name, undefined, { urn });
            case "azure-native:mobilepacketcore/v20231015preview:ObservabilityService":
                return new exports.ObservabilityService(name, undefined, { urn });
            case "azure-native:mobilepacketcore/v20231015preview:SmfDeployment":
                return new exports.SmfDeployment(name, undefined, { urn });
            case "azure-native:mobilepacketcore/v20231015preview:UpfDeployment":
                return new exports.UpfDeployment(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "mobilepacketcore/v20231015preview", _module);
