// @pulumi/azure-native/app/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getJavaComponent = exports.getDotNetComponentOutput = exports.getDotNetComponent = exports.getDaprSubscriptionOutput = exports.getDaprSubscription = exports.getDaprComponentResiliencyPolicyOutput = exports.getDaprComponentResiliencyPolicy = exports.getDaprComponentOutput = exports.getDaprComponent = exports.getCustomDomainVerificationIdOutput = exports.getCustomDomainVerificationId = exports.getContainerAppsSourceControlOutput = exports.getContainerAppsSourceControl = exports.getContainerAppsAuthConfigOutput = exports.getContainerAppsAuthConfig = exports.getContainerAppAuthTokenOutput = exports.getContainerAppAuthToken = exports.getContainerAppOutput = exports.getContainerApp = exports.getConnectedEnvironmentsStorageOutput = exports.getConnectedEnvironmentsStorage = exports.getConnectedEnvironmentsDaprComponentOutput = exports.getConnectedEnvironmentsDaprComponent = exports.getConnectedEnvironmentsCertificateOutput = exports.getConnectedEnvironmentsCertificate = exports.getConnectedEnvironmentOutput = exports.getConnectedEnvironment = exports.getCertificateOutput = exports.getCertificate = exports.getBuilderOutput = exports.getBuilder = exports.getBuildOutput = exports.getBuild = exports.getAppResiliencyOutput = exports.getAppResiliency = exports.DotNetComponent = exports.DaprSubscription = exports.DaprComponentResiliencyPolicy = exports.DaprComponent = exports.ContainerAppsSourceControl = exports.ContainerAppsAuthConfig = exports.ContainerApp = exports.ConnectedEnvironmentsStorage = exports.ConnectedEnvironmentsDaprComponent = exports.ConnectedEnvironmentsCertificate = exports.ConnectedEnvironment = exports.Certificate = exports.Builder = exports.Build = exports.AppResiliency = void 0;
exports.v20240301 = exports.v20231102preview = exports.v20230801preview = exports.v20230502preview = exports.v20230501 = exports.v20230401preview = exports.v20221001 = exports.v20220101preview = exports.ManagedEnvironmentsStorage = exports.ManagedEnvironment = exports.ManagedCertificate = exports.listJobSecretsOutput = exports.listJobSecrets = exports.listDaprComponentSecretsOutput = exports.listDaprComponentSecrets = exports.listContainerAppSecretsOutput = exports.listContainerAppSecrets = exports.listContainerAppCustomHostNameAnalysisOutput = exports.listContainerAppCustomHostNameAnalysis = exports.listConnectedEnvironmentsDaprComponentSecretsOutput = exports.listConnectedEnvironmentsDaprComponentSecrets = exports.listBuildAuthTokenOutput = exports.listBuildAuthToken = exports.Job = exports.JavaComponent = exports.getManagedEnvironmentsStorageOutput = exports.getManagedEnvironmentsStorage = exports.getManagedEnvironmentAuthTokenOutput = exports.getManagedEnvironmentAuthToken = exports.getManagedEnvironmentOutput = exports.getManagedEnvironment = exports.getManagedCertificateOutput = exports.getManagedCertificate = exports.getJobOutput = exports.getJob = exports.getJavaComponentOutput = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AppResiliency = null;
utilities.lazyLoad(exports, ["AppResiliency"], () => require("./appResiliency"));
exports.Build = null;
utilities.lazyLoad(exports, ["Build"], () => require("./build"));
exports.Builder = null;
utilities.lazyLoad(exports, ["Builder"], () => require("./builder"));
exports.Certificate = null;
utilities.lazyLoad(exports, ["Certificate"], () => require("./certificate"));
exports.ConnectedEnvironment = null;
utilities.lazyLoad(exports, ["ConnectedEnvironment"], () => require("./connectedEnvironment"));
exports.ConnectedEnvironmentsCertificate = null;
utilities.lazyLoad(exports, ["ConnectedEnvironmentsCertificate"], () => require("./connectedEnvironmentsCertificate"));
exports.ConnectedEnvironmentsDaprComponent = null;
utilities.lazyLoad(exports, ["ConnectedEnvironmentsDaprComponent"], () => require("./connectedEnvironmentsDaprComponent"));
exports.ConnectedEnvironmentsStorage = null;
utilities.lazyLoad(exports, ["ConnectedEnvironmentsStorage"], () => require("./connectedEnvironmentsStorage"));
exports.ContainerApp = null;
utilities.lazyLoad(exports, ["ContainerApp"], () => require("./containerApp"));
exports.ContainerAppsAuthConfig = null;
utilities.lazyLoad(exports, ["ContainerAppsAuthConfig"], () => require("./containerAppsAuthConfig"));
exports.ContainerAppsSourceControl = null;
utilities.lazyLoad(exports, ["ContainerAppsSourceControl"], () => require("./containerAppsSourceControl"));
exports.DaprComponent = null;
utilities.lazyLoad(exports, ["DaprComponent"], () => require("./daprComponent"));
exports.DaprComponentResiliencyPolicy = null;
utilities.lazyLoad(exports, ["DaprComponentResiliencyPolicy"], () => require("./daprComponentResiliencyPolicy"));
exports.DaprSubscription = null;
utilities.lazyLoad(exports, ["DaprSubscription"], () => require("./daprSubscription"));
exports.DotNetComponent = null;
utilities.lazyLoad(exports, ["DotNetComponent"], () => require("./dotNetComponent"));
exports.getAppResiliency = null;
exports.getAppResiliencyOutput = null;
utilities.lazyLoad(exports, ["getAppResiliency", "getAppResiliencyOutput"], () => require("./getAppResiliency"));
exports.getBuild = null;
exports.getBuildOutput = null;
utilities.lazyLoad(exports, ["getBuild", "getBuildOutput"], () => require("./getBuild"));
exports.getBuilder = null;
exports.getBuilderOutput = null;
utilities.lazyLoad(exports, ["getBuilder", "getBuilderOutput"], () => require("./getBuilder"));
exports.getCertificate = null;
exports.getCertificateOutput = null;
utilities.lazyLoad(exports, ["getCertificate", "getCertificateOutput"], () => require("./getCertificate"));
exports.getConnectedEnvironment = null;
exports.getConnectedEnvironmentOutput = null;
utilities.lazyLoad(exports, ["getConnectedEnvironment", "getConnectedEnvironmentOutput"], () => require("./getConnectedEnvironment"));
exports.getConnectedEnvironmentsCertificate = null;
exports.getConnectedEnvironmentsCertificateOutput = null;
utilities.lazyLoad(exports, ["getConnectedEnvironmentsCertificate", "getConnectedEnvironmentsCertificateOutput"], () => require("./getConnectedEnvironmentsCertificate"));
exports.getConnectedEnvironmentsDaprComponent = null;
exports.getConnectedEnvironmentsDaprComponentOutput = null;
utilities.lazyLoad(exports, ["getConnectedEnvironmentsDaprComponent", "getConnectedEnvironmentsDaprComponentOutput"], () => require("./getConnectedEnvironmentsDaprComponent"));
exports.getConnectedEnvironmentsStorage = null;
exports.getConnectedEnvironmentsStorageOutput = null;
utilities.lazyLoad(exports, ["getConnectedEnvironmentsStorage", "getConnectedEnvironmentsStorageOutput"], () => require("./getConnectedEnvironmentsStorage"));
exports.getContainerApp = null;
exports.getContainerAppOutput = null;
utilities.lazyLoad(exports, ["getContainerApp", "getContainerAppOutput"], () => require("./getContainerApp"));
exports.getContainerAppAuthToken = null;
exports.getContainerAppAuthTokenOutput = null;
utilities.lazyLoad(exports, ["getContainerAppAuthToken", "getContainerAppAuthTokenOutput"], () => require("./getContainerAppAuthToken"));
exports.getContainerAppsAuthConfig = null;
exports.getContainerAppsAuthConfigOutput = null;
utilities.lazyLoad(exports, ["getContainerAppsAuthConfig", "getContainerAppsAuthConfigOutput"], () => require("./getContainerAppsAuthConfig"));
exports.getContainerAppsSourceControl = null;
exports.getContainerAppsSourceControlOutput = null;
utilities.lazyLoad(exports, ["getContainerAppsSourceControl", "getContainerAppsSourceControlOutput"], () => require("./getContainerAppsSourceControl"));
exports.getCustomDomainVerificationId = null;
exports.getCustomDomainVerificationIdOutput = null;
utilities.lazyLoad(exports, ["getCustomDomainVerificationId", "getCustomDomainVerificationIdOutput"], () => require("./getCustomDomainVerificationId"));
exports.getDaprComponent = null;
exports.getDaprComponentOutput = null;
utilities.lazyLoad(exports, ["getDaprComponent", "getDaprComponentOutput"], () => require("./getDaprComponent"));
exports.getDaprComponentResiliencyPolicy = null;
exports.getDaprComponentResiliencyPolicyOutput = null;
utilities.lazyLoad(exports, ["getDaprComponentResiliencyPolicy", "getDaprComponentResiliencyPolicyOutput"], () => require("./getDaprComponentResiliencyPolicy"));
exports.getDaprSubscription = null;
exports.getDaprSubscriptionOutput = null;
utilities.lazyLoad(exports, ["getDaprSubscription", "getDaprSubscriptionOutput"], () => require("./getDaprSubscription"));
exports.getDotNetComponent = null;
exports.getDotNetComponentOutput = null;
utilities.lazyLoad(exports, ["getDotNetComponent", "getDotNetComponentOutput"], () => require("./getDotNetComponent"));
exports.getJavaComponent = null;
exports.getJavaComponentOutput = null;
utilities.lazyLoad(exports, ["getJavaComponent", "getJavaComponentOutput"], () => require("./getJavaComponent"));
exports.getJob = null;
exports.getJobOutput = null;
utilities.lazyLoad(exports, ["getJob", "getJobOutput"], () => require("./getJob"));
exports.getManagedCertificate = null;
exports.getManagedCertificateOutput = null;
utilities.lazyLoad(exports, ["getManagedCertificate", "getManagedCertificateOutput"], () => require("./getManagedCertificate"));
exports.getManagedEnvironment = null;
exports.getManagedEnvironmentOutput = null;
utilities.lazyLoad(exports, ["getManagedEnvironment", "getManagedEnvironmentOutput"], () => require("./getManagedEnvironment"));
exports.getManagedEnvironmentAuthToken = null;
exports.getManagedEnvironmentAuthTokenOutput = null;
utilities.lazyLoad(exports, ["getManagedEnvironmentAuthToken", "getManagedEnvironmentAuthTokenOutput"], () => require("./getManagedEnvironmentAuthToken"));
exports.getManagedEnvironmentsStorage = null;
exports.getManagedEnvironmentsStorageOutput = null;
utilities.lazyLoad(exports, ["getManagedEnvironmentsStorage", "getManagedEnvironmentsStorageOutput"], () => require("./getManagedEnvironmentsStorage"));
exports.JavaComponent = null;
utilities.lazyLoad(exports, ["JavaComponent"], () => require("./javaComponent"));
exports.Job = null;
utilities.lazyLoad(exports, ["Job"], () => require("./job"));
exports.listBuildAuthToken = null;
exports.listBuildAuthTokenOutput = null;
utilities.lazyLoad(exports, ["listBuildAuthToken", "listBuildAuthTokenOutput"], () => require("./listBuildAuthToken"));
exports.listConnectedEnvironmentsDaprComponentSecrets = null;
exports.listConnectedEnvironmentsDaprComponentSecretsOutput = null;
utilities.lazyLoad(exports, ["listConnectedEnvironmentsDaprComponentSecrets", "listConnectedEnvironmentsDaprComponentSecretsOutput"], () => require("./listConnectedEnvironmentsDaprComponentSecrets"));
exports.listContainerAppCustomHostNameAnalysis = null;
exports.listContainerAppCustomHostNameAnalysisOutput = null;
utilities.lazyLoad(exports, ["listContainerAppCustomHostNameAnalysis", "listContainerAppCustomHostNameAnalysisOutput"], () => require("./listContainerAppCustomHostNameAnalysis"));
exports.listContainerAppSecrets = null;
exports.listContainerAppSecretsOutput = null;
utilities.lazyLoad(exports, ["listContainerAppSecrets", "listContainerAppSecretsOutput"], () => require("./listContainerAppSecrets"));
exports.listDaprComponentSecrets = null;
exports.listDaprComponentSecretsOutput = null;
utilities.lazyLoad(exports, ["listDaprComponentSecrets", "listDaprComponentSecretsOutput"], () => require("./listDaprComponentSecrets"));
exports.listJobSecrets = null;
exports.listJobSecretsOutput = null;
utilities.lazyLoad(exports, ["listJobSecrets", "listJobSecretsOutput"], () => require("./listJobSecrets"));
exports.ManagedCertificate = null;
utilities.lazyLoad(exports, ["ManagedCertificate"], () => require("./managedCertificate"));
exports.ManagedEnvironment = null;
utilities.lazyLoad(exports, ["ManagedEnvironment"], () => require("./managedEnvironment"));
exports.ManagedEnvironmentsStorage = null;
utilities.lazyLoad(exports, ["ManagedEnvironmentsStorage"], () => require("./managedEnvironmentsStorage"));
// Export enums:
__exportStar(require("../types/enums/app"), exports);
// Export sub-modules:
const v20220101preview = require("./v20220101preview");
exports.v20220101preview = v20220101preview;
const v20221001 = require("./v20221001");
exports.v20221001 = v20221001;
const v20230401preview = require("./v20230401preview");
exports.v20230401preview = v20230401preview;
const v20230501 = require("./v20230501");
exports.v20230501 = v20230501;
const v20230502preview = require("./v20230502preview");
exports.v20230502preview = v20230502preview;
const v20230801preview = require("./v20230801preview");
exports.v20230801preview = v20230801preview;
const v20231102preview = require("./v20231102preview");
exports.v20231102preview = v20231102preview;
const v20240301 = require("./v20240301");
exports.v20240301 = v20240301;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:app:AppResiliency":
                return new exports.AppResiliency(name, undefined, { urn });
            case "azure-native:app:Build":
                return new exports.Build(name, undefined, { urn });
            case "azure-native:app:Builder":
                return new exports.Builder(name, undefined, { urn });
            case "azure-native:app:Certificate":
                return new exports.Certificate(name, undefined, { urn });
            case "azure-native:app:ConnectedEnvironment":
                return new exports.ConnectedEnvironment(name, undefined, { urn });
            case "azure-native:app:ConnectedEnvironmentsCertificate":
                return new exports.ConnectedEnvironmentsCertificate(name, undefined, { urn });
            case "azure-native:app:ConnectedEnvironmentsDaprComponent":
                return new exports.ConnectedEnvironmentsDaprComponent(name, undefined, { urn });
            case "azure-native:app:ConnectedEnvironmentsStorage":
                return new exports.ConnectedEnvironmentsStorage(name, undefined, { urn });
            case "azure-native:app:ContainerApp":
                return new exports.ContainerApp(name, undefined, { urn });
            case "azure-native:app:ContainerAppsAuthConfig":
                return new exports.ContainerAppsAuthConfig(name, undefined, { urn });
            case "azure-native:app:ContainerAppsSourceControl":
                return new exports.ContainerAppsSourceControl(name, undefined, { urn });
            case "azure-native:app:DaprComponent":
                return new exports.DaprComponent(name, undefined, { urn });
            case "azure-native:app:DaprComponentResiliencyPolicy":
                return new exports.DaprComponentResiliencyPolicy(name, undefined, { urn });
            case "azure-native:app:DaprSubscription":
                return new exports.DaprSubscription(name, undefined, { urn });
            case "azure-native:app:DotNetComponent":
                return new exports.DotNetComponent(name, undefined, { urn });
            case "azure-native:app:JavaComponent":
                return new exports.JavaComponent(name, undefined, { urn });
            case "azure-native:app:Job":
                return new exports.Job(name, undefined, { urn });
            case "azure-native:app:ManagedCertificate":
                return new exports.ManagedCertificate(name, undefined, { urn });
            case "azure-native:app:ManagedEnvironment":
                return new exports.ManagedEnvironment(name, undefined, { urn });
            case "azure-native:app:ManagedEnvironmentsStorage":
                return new exports.ManagedEnvironmentsStorage(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "app", _module);
