// @pulumi/azure-native/containerservice/openShiftManagedCluster.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.OpenShiftManagedCluster = void 0;
const pulumi = require("@pulumi/pulumi");
const inputs = require("../types/input");
const utilities = require("../utilities");
/**
 * OpenShift Managed cluster.
 * Azure REST API version: 2019-10-27-preview. Prior API version in Azure Native 1.x: 2019-04-30.
 *
 * Other available API versions: 2018-09-30-preview.
 */
class OpenShiftManagedCluster extends pulumi.CustomResource {
    /**
     * Get an existing OpenShiftManagedCluster resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new OpenShiftManagedCluster(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of OpenShiftManagedCluster.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === OpenShiftManagedCluster.__pulumiType;
    }
    /**
     * Create a OpenShiftManagedCluster resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.openShiftVersion === undefined) && !opts.urn) {
                throw new Error("Missing required property 'openShiftVersion'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["agentPoolProfiles"] = args ? args.agentPoolProfiles : undefined;
            resourceInputs["authProfile"] = args ? args.authProfile : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["masterPoolProfile"] = args ? args.masterPoolProfile : undefined;
            resourceInputs["monitorProfile"] = args ? args.monitorProfile : undefined;
            resourceInputs["networkProfile"] = args ? (args.networkProfile ? pulumi.output(args.networkProfile).apply(inputs.containerservice.networkProfileArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["openShiftVersion"] = args ? args.openShiftVersion : undefined;
            resourceInputs["plan"] = args ? args.plan : undefined;
            resourceInputs["refreshCluster"] = args ? args.refreshCluster : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["resourceName"] = args ? args.resourceName : undefined;
            resourceInputs["routerProfiles"] = args ? args.routerProfiles : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["clusterVersion"] = undefined /*out*/;
            resourceInputs["fqdn"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["publicHostname"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["agentPoolProfiles"] = undefined /*out*/;
            resourceInputs["authProfile"] = undefined /*out*/;
            resourceInputs["clusterVersion"] = undefined /*out*/;
            resourceInputs["fqdn"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["masterPoolProfile"] = undefined /*out*/;
            resourceInputs["monitorProfile"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["networkProfile"] = undefined /*out*/;
            resourceInputs["openShiftVersion"] = undefined /*out*/;
            resourceInputs["plan"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["publicHostname"] = undefined /*out*/;
            resourceInputs["refreshCluster"] = undefined /*out*/;
            resourceInputs["routerProfiles"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:containerservice/v20180930preview:OpenShiftManagedCluster" }, { type: "azure-native:containerservice/v20190430:OpenShiftManagedCluster" }, { type: "azure-native:containerservice/v20190930preview:OpenShiftManagedCluster" }, { type: "azure-native:containerservice/v20191027preview:OpenShiftManagedCluster" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(OpenShiftManagedCluster.__pulumiType, name, resourceInputs, opts);
    }
}
exports.OpenShiftManagedCluster = OpenShiftManagedCluster;
/** @internal */
OpenShiftManagedCluster.__pulumiType = 'azure-native:containerservice:OpenShiftManagedCluster';
