// @pulumi/azure-native/web/v20201201/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAppScmAllowed = exports.WebAppFtpAllowed = exports.listSiteIdentifiersAssignedToHostNameOutput = exports.listSiteIdentifiersAssignedToHostName = exports.getWebAppScmAllowedOutput = exports.getWebAppScmAllowed = exports.getWebAppFtpAllowedOutput = exports.getWebAppFtpAllowed = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.getWebAppFtpAllowed = null;
exports.getWebAppFtpAllowedOutput = null;
utilities.lazyLoad(exports, ["getWebAppFtpAllowed", "getWebAppFtpAllowedOutput"], () => require("./getWebAppFtpAllowed"));
exports.getWebAppScmAllowed = null;
exports.getWebAppScmAllowedOutput = null;
utilities.lazyLoad(exports, ["getWebAppScmAllowed", "getWebAppScmAllowedOutput"], () => require("./getWebAppScmAllowed"));
exports.listSiteIdentifiersAssignedToHostName = null;
exports.listSiteIdentifiersAssignedToHostNameOutput = null;
utilities.lazyLoad(exports, ["listSiteIdentifiersAssignedToHostName", "listSiteIdentifiersAssignedToHostNameOutput"], () => require("./listSiteIdentifiersAssignedToHostName"));
exports.WebAppFtpAllowed = null;
utilities.lazyLoad(exports, ["WebAppFtpAllowed"], () => require("./webAppFtpAllowed"));
exports.WebAppScmAllowed = null;
utilities.lazyLoad(exports, ["WebAppScmAllowed"], () => require("./webAppScmAllowed"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:web/v20201201:WebAppFtpAllowed":
                return new exports.WebAppFtpAllowed(name, undefined, { urn });
            case "azure-native:web/v20201201:WebAppScmAllowed":
                return new exports.WebAppScmAllowed(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "web/v20201201", _module);
