// @pulumi/azure-native/blockchain/blockchainMember.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.BlockchainMember = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Payload of the blockchain member which is exposed in the request/response of the resource provider.
 * Azure REST API version: 2018-06-01-preview. Prior API version in Azure Native 1.x: 2018-06-01-preview.
 */
class BlockchainMember extends pulumi.CustomResource {
    /**
     * Get an existing BlockchainMember resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new BlockchainMember(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of BlockchainMember.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === BlockchainMember.__pulumiType;
    }
    /**
     * Create a BlockchainMember resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["blockchainMemberName"] = args ? args.blockchainMemberName : undefined;
            resourceInputs["consortium"] = args ? args.consortium : undefined;
            resourceInputs["consortiumManagementAccountPassword"] = args ? args.consortiumManagementAccountPassword : undefined;
            resourceInputs["consortiumMemberDisplayName"] = args ? args.consortiumMemberDisplayName : undefined;
            resourceInputs["consortiumRole"] = args ? args.consortiumRole : undefined;
            resourceInputs["firewallRules"] = args ? args.firewallRules : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["password"] = args ? args.password : undefined;
            resourceInputs["protocol"] = args ? args.protocol : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["sku"] = args ? args.sku : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["validatorNodesSku"] = args ? args.validatorNodesSku : undefined;
            resourceInputs["consortiumManagementAccountAddress"] = undefined /*out*/;
            resourceInputs["dns"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["publicKey"] = undefined /*out*/;
            resourceInputs["rootContractAddress"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["userName"] = undefined /*out*/;
        }
        else {
            resourceInputs["consortium"] = undefined /*out*/;
            resourceInputs["consortiumManagementAccountAddress"] = undefined /*out*/;
            resourceInputs["consortiumManagementAccountPassword"] = undefined /*out*/;
            resourceInputs["consortiumMemberDisplayName"] = undefined /*out*/;
            resourceInputs["consortiumRole"] = undefined /*out*/;
            resourceInputs["dns"] = undefined /*out*/;
            resourceInputs["firewallRules"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["password"] = undefined /*out*/;
            resourceInputs["protocol"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["publicKey"] = undefined /*out*/;
            resourceInputs["rootContractAddress"] = undefined /*out*/;
            resourceInputs["sku"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["userName"] = undefined /*out*/;
            resourceInputs["validatorNodesSku"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:blockchain/v20180601preview:BlockchainMember" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(BlockchainMember.__pulumiType, name, resourceInputs, opts);
    }
}
exports.BlockchainMember = BlockchainMember;
/** @internal */
BlockchainMember.__pulumiType = 'azure-native:blockchain:BlockchainMember';
