// @pulumi/azure-native/blockchain/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20180601preview = exports.TransactionNode = exports.listTransactionNodeApiKeysOutput = exports.listTransactionNodeApiKeys = exports.listLocationConsortiumsOutput = exports.listLocationConsortiums = exports.listBlockchainMemberApiKeysOutput = exports.listBlockchainMemberApiKeys = exports.getTransactionNodeOutput = exports.getTransactionNode = exports.getBlockchainMemberOutput = exports.getBlockchainMember = exports.BlockchainMember = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.BlockchainMember = null;
utilities.lazyLoad(exports, ["BlockchainMember"], () => require("./blockchainMember"));
exports.getBlockchainMember = null;
exports.getBlockchainMemberOutput = null;
utilities.lazyLoad(exports, ["getBlockchainMember", "getBlockchainMemberOutput"], () => require("./getBlockchainMember"));
exports.getTransactionNode = null;
exports.getTransactionNodeOutput = null;
utilities.lazyLoad(exports, ["getTransactionNode", "getTransactionNodeOutput"], () => require("./getTransactionNode"));
exports.listBlockchainMemberApiKeys = null;
exports.listBlockchainMemberApiKeysOutput = null;
utilities.lazyLoad(exports, ["listBlockchainMemberApiKeys", "listBlockchainMemberApiKeysOutput"], () => require("./listBlockchainMemberApiKeys"));
exports.listLocationConsortiums = null;
exports.listLocationConsortiumsOutput = null;
utilities.lazyLoad(exports, ["listLocationConsortiums", "listLocationConsortiumsOutput"], () => require("./listLocationConsortiums"));
exports.listTransactionNodeApiKeys = null;
exports.listTransactionNodeApiKeysOutput = null;
utilities.lazyLoad(exports, ["listTransactionNodeApiKeys", "listTransactionNodeApiKeysOutput"], () => require("./listTransactionNodeApiKeys"));
exports.TransactionNode = null;
utilities.lazyLoad(exports, ["TransactionNode"], () => require("./transactionNode"));
// Export enums:
__exportStar(require("../types/enums/blockchain"), exports);
// Export sub-modules:
const v20180601preview = require("./v20180601preview");
exports.v20180601preview = v20180601preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:blockchain:BlockchainMember":
                return new exports.BlockchainMember(name, undefined, { urn });
            case "azure-native:blockchain:TransactionNode":
                return new exports.TransactionNode(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "blockchain", _module);
