// @pulumi/azure-native/insights/v20160301/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LogProfile = exports.getLogProfileOutput = exports.getLogProfile = exports.getAlertRuleOutput = exports.getAlertRule = exports.AlertRule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.AlertRule = null;
utilities.lazyLoad(exports, ["AlertRule"], () => require("./alertRule"));
exports.getAlertRule = null;
exports.getAlertRuleOutput = null;
utilities.lazyLoad(exports, ["getAlertRule", "getAlertRuleOutput"], () => require("./getAlertRule"));
exports.getLogProfile = null;
exports.getLogProfileOutput = null;
utilities.lazyLoad(exports, ["getLogProfile", "getLogProfileOutput"], () => require("./getLogProfile"));
exports.LogProfile = null;
utilities.lazyLoad(exports, ["LogProfile"], () => require("./logProfile"));
// Export enums:
__exportStar(require("../../types/enums/insights/v20160301"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:insights/v20160301:AlertRule":
                return new exports.AlertRule(name, undefined, { urn });
            case "azure-native:insights/v20160301:LogProfile":
                return new exports.LogProfile(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "insights/v20160301", _module);
