// @pulumi/azure-native/monitor/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20230403 = exports.getAzureMonitorWorkspaceOutput = exports.getAzureMonitorWorkspace = exports.AzureMonitorWorkspace = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AzureMonitorWorkspace = null;
utilities.lazyLoad(exports, ["AzureMonitorWorkspace"], () => require("./azureMonitorWorkspace"));
exports.getAzureMonitorWorkspace = null;
exports.getAzureMonitorWorkspaceOutput = null;
utilities.lazyLoad(exports, ["getAzureMonitorWorkspace", "getAzureMonitorWorkspaceOutput"], () => require("./getAzureMonitorWorkspace"));
// Export sub-modules:
const v20230403 = require("./v20230403");
exports.v20230403 = v20230403;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:monitor:AzureMonitorWorkspace":
                return new exports.AzureMonitorWorkspace(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "monitor", _module);
