// @pulumi/azure-native/azuredatatransfer/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20240125 = exports.v20231011preview = exports.Pipeline = exports.listListSchemaOutput = exports.listListSchema = exports.listListPendingFlowOutput = exports.listListPendingFlow = exports.listListPendingConnectionOutput = exports.listListPendingConnection = exports.listAzureDataTransferApprovedSchemasOutput = exports.listAzureDataTransferApprovedSchemas = exports.getPipelineOutput = exports.getPipeline = exports.getFlowOutput = exports.getFlow = exports.getConnectionOutput = exports.getConnection = exports.Flow = exports.Connection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Connection = null;
utilities.lazyLoad(exports, ["Connection"], () => require("./connection"));
exports.Flow = null;
utilities.lazyLoad(exports, ["Flow"], () => require("./flow"));
exports.getConnection = null;
exports.getConnectionOutput = null;
utilities.lazyLoad(exports, ["getConnection", "getConnectionOutput"], () => require("./getConnection"));
exports.getFlow = null;
exports.getFlowOutput = null;
utilities.lazyLoad(exports, ["getFlow", "getFlowOutput"], () => require("./getFlow"));
exports.getPipeline = null;
exports.getPipelineOutput = null;
utilities.lazyLoad(exports, ["getPipeline", "getPipelineOutput"], () => require("./getPipeline"));
exports.listAzureDataTransferApprovedSchemas = null;
exports.listAzureDataTransferApprovedSchemasOutput = null;
utilities.lazyLoad(exports, ["listAzureDataTransferApprovedSchemas", "listAzureDataTransferApprovedSchemasOutput"], () => require("./listAzureDataTransferApprovedSchemas"));
exports.listListPendingConnection = null;
exports.listListPendingConnectionOutput = null;
utilities.lazyLoad(exports, ["listListPendingConnection", "listListPendingConnectionOutput"], () => require("./listListPendingConnection"));
exports.listListPendingFlow = null;
exports.listListPendingFlowOutput = null;
utilities.lazyLoad(exports, ["listListPendingFlow", "listListPendingFlowOutput"], () => require("./listListPendingFlow"));
exports.listListSchema = null;
exports.listListSchemaOutput = null;
utilities.lazyLoad(exports, ["listListSchema", "listListSchemaOutput"], () => require("./listListSchema"));
exports.Pipeline = null;
utilities.lazyLoad(exports, ["Pipeline"], () => require("./pipeline"));
// Export enums:
__exportStar(require("../types/enums/azuredatatransfer"), exports);
// Export sub-modules:
const v20231011preview = require("./v20231011preview");
exports.v20231011preview = v20231011preview;
const v20240125 = require("./v20240125");
exports.v20240125 = v20240125;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:azuredatatransfer:Connection":
                return new exports.Connection(name, undefined, { urn });
            case "azure-native:azuredatatransfer:Flow":
                return new exports.Flow(name, undefined, { urn });
            case "azure-native:azuredatatransfer:Pipeline":
                return new exports.Pipeline(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "azuredatatransfer", _module);
