// @pulumi/azure-native/containerinstance/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20230501 = exports.v20230201preview = exports.v20210701 = exports.v20210301 = exports.getContainerGroupOutput = exports.getContainerGroup = exports.ContainerGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.ContainerGroup = null;
utilities.lazyLoad(exports, ["ContainerGroup"], () => require("./containerGroup"));
exports.getContainerGroup = null;
exports.getContainerGroupOutput = null;
utilities.lazyLoad(exports, ["getContainerGroup", "getContainerGroupOutput"], () => require("./getContainerGroup"));
// Export enums:
__exportStar(require("../types/enums/containerinstance"), exports);
// Export sub-modules:
const v20210301 = require("./v20210301");
exports.v20210301 = v20210301;
const v20210701 = require("./v20210701");
exports.v20210701 = v20210701;
const v20230201preview = require("./v20230201preview");
exports.v20230201preview = v20230201preview;
const v20230501 = require("./v20230501");
exports.v20230501 = v20230501;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:containerinstance:ContainerGroup":
                return new exports.ContainerGroup(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "containerinstance", _module);
