// @pulumi/azure-native/securitydevops/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20220901preview = exports.GitHubConnector = exports.getGitHubConnectorOutput = exports.getGitHubConnector = exports.getAzureDevOpsConnectorOutput = exports.getAzureDevOpsConnector = exports.AzureDevOpsConnector = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AzureDevOpsConnector = null;
utilities.lazyLoad(exports, ["AzureDevOpsConnector"], () => require("./azureDevOpsConnector"));
exports.getAzureDevOpsConnector = null;
exports.getAzureDevOpsConnectorOutput = null;
utilities.lazyLoad(exports, ["getAzureDevOpsConnector", "getAzureDevOpsConnectorOutput"], () => require("./getAzureDevOpsConnector"));
exports.getGitHubConnector = null;
exports.getGitHubConnectorOutput = null;
utilities.lazyLoad(exports, ["getGitHubConnector", "getGitHubConnectorOutput"], () => require("./getGitHubConnector"));
exports.GitHubConnector = null;
utilities.lazyLoad(exports, ["GitHubConnector"], () => require("./gitHubConnector"));
// Export enums:
__exportStar(require("../types/enums/securitydevops"), exports);
// Export sub-modules:
const v20220901preview = require("./v20220901preview");
exports.v20220901preview = v20220901preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:securitydevops:AzureDevOpsConnector":
                return new exports.AzureDevOpsConnector(name, undefined, { urn });
            case "azure-native:securitydevops:GitHubConnector":
                return new exports.GitHubConnector(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "securitydevops", _module);
