// @pulumi/azure-native/alertsmanagement/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20230501preview = exports.v20230401preview = exports.v20230301 = exports.v20210808 = exports.v20210401 = exports.v20190505preview = exports.SmartDetectorAlertRule = exports.PrometheusRuleGroup = exports.getSmartDetectorAlertRuleOutput = exports.getSmartDetectorAlertRule = exports.getPrometheusRuleGroupOutput = exports.getPrometheusRuleGroup = exports.getAlertProcessingRuleByNameOutput = exports.getAlertProcessingRuleByName = exports.getActionRuleByNameOutput = exports.getActionRuleByName = exports.AlertProcessingRuleByName = exports.ActionRuleByName = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.ActionRuleByName = null;
utilities.lazyLoad(exports, ["ActionRuleByName"], () => require("./actionRuleByName"));
exports.AlertProcessingRuleByName = null;
utilities.lazyLoad(exports, ["AlertProcessingRuleByName"], () => require("./alertProcessingRuleByName"));
exports.getActionRuleByName = null;
exports.getActionRuleByNameOutput = null;
utilities.lazyLoad(exports, ["getActionRuleByName", "getActionRuleByNameOutput"], () => require("./getActionRuleByName"));
exports.getAlertProcessingRuleByName = null;
exports.getAlertProcessingRuleByNameOutput = null;
utilities.lazyLoad(exports, ["getAlertProcessingRuleByName", "getAlertProcessingRuleByNameOutput"], () => require("./getAlertProcessingRuleByName"));
exports.getPrometheusRuleGroup = null;
exports.getPrometheusRuleGroupOutput = null;
utilities.lazyLoad(exports, ["getPrometheusRuleGroup", "getPrometheusRuleGroupOutput"], () => require("./getPrometheusRuleGroup"));
exports.getSmartDetectorAlertRule = null;
exports.getSmartDetectorAlertRuleOutput = null;
utilities.lazyLoad(exports, ["getSmartDetectorAlertRule", "getSmartDetectorAlertRuleOutput"], () => require("./getSmartDetectorAlertRule"));
exports.PrometheusRuleGroup = null;
utilities.lazyLoad(exports, ["PrometheusRuleGroup"], () => require("./prometheusRuleGroup"));
exports.SmartDetectorAlertRule = null;
utilities.lazyLoad(exports, ["SmartDetectorAlertRule"], () => require("./smartDetectorAlertRule"));
// Export enums:
__exportStar(require("../types/enums/alertsmanagement"), exports);
// Export sub-modules:
const v20190505preview = require("./v20190505preview");
exports.v20190505preview = v20190505preview;
const v20210401 = require("./v20210401");
exports.v20210401 = v20210401;
const v20210808 = require("./v20210808");
exports.v20210808 = v20210808;
const v20230301 = require("./v20230301");
exports.v20230301 = v20230301;
const v20230401preview = require("./v20230401preview");
exports.v20230401preview = v20230401preview;
const v20230501preview = require("./v20230501preview");
exports.v20230501preview = v20230501preview;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:alertsmanagement:ActionRuleByName":
                return new exports.ActionRuleByName(name, undefined, { urn });
            case "azure-native:alertsmanagement:AlertProcessingRuleByName":
                return new exports.AlertProcessingRuleByName(name, undefined, { urn });
            case "azure-native:alertsmanagement:PrometheusRuleGroup":
                return new exports.PrometheusRuleGroup(name, undefined, { urn });
            case "azure-native:alertsmanagement:SmartDetectorAlertRule":
                return new exports.SmartDetectorAlertRule(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "alertsmanagement", _module);
